core/synchronizer.go

```diff
@@ -645,14 +641,15 @@ func (s *Synchronizer) sendPatchObject(ctx context.Context, id domain.KindName,
 
 func (s *Synchronizer) sendPing(ctx context.Context) {
 	event := domain.EventPing
-	msg := domain.Generic{
-		Event: &event,
-	}
-	data, err := json.Marshal(msg)
-	if err != nil {
-		logger.L().Fatal("marshal ping message", helpers.Error(err))
-	}
 	for {
+		msg := domain.Generic{
+			Event: &event,
+			MsgId: utils.NewMsgId(),
+		}
+		data, err := json.Marshal(msg)
+		if err != nil {
+			logger.L().Fatal("marshal ping message", helpers.Error(err))
```

Comment:
Is this a client-server flow or just a client? Is fatal the correct way to handle the error?

Reply:
this is client only - it depends but we want to make sure we are sending those pings
on the other hand if we cannot marshal a json we should probably quit?